### PR TITLE
[lib] Update Misc with 4.08.1 stdlib updates

### PR DESCRIPTION
**Pull request by HadrienRenaud**
This is neither urgent nor very important for me, but I somehow did it.

### CODE REVIEW

File: lib/misc.ml [hunk 1]
```diff
@@ -36,77 +36,43 @@ let dot_name name_xxx =
...
-let seq_opt f = function
-  | Some x -> f x
-  | None -> None
-
-let app_opt f = function
-  | None -> None
-  | Some x -> Some (f x)
+let is_none = Option.is_none
+let is_some = Option.is_some
+let as_some = Option.get
```
Comment (relokin) [inline lib/misc.ml:61]: This changes the behavior when its argument is None from assert to raise Invalid_argument but I think this makes more sense.

File: lib/misc.ml [hunk 2]
```diff
@@ -165,79 +122,10 @@ let rec list_eq eq xs ys = match xs,ys with
...
-
-let capitalize s = match s with
-| "" -> assert false
-| _ ->
-    String.make 1 (char_uppercase s.[0]) ^
-    String.sub s 1 (String.length s-1)
+let char_uppercase = Char.uppercase_ascii
+let lowercase = String.lowercase_ascii
+let uppercase = String.uppercase_ascii
+let capitalize = String.capitalize_ascii
```
Comment (relokin) [inline lib/misc.ml:128]: Slightly different behaviour in the case of
```
Misc.capitalize ""
```
used to assert now it returns ""
Again the new behaviour makes sense to me, just checking.

**Review by relokin — APPROVED** [review]
LGTM!

**Comment by maranget** [comment]
Merged, thanks

**Pull request merged**